com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -187,28 +187,50 @@ internal void RemoveOwnership(NetworkObject networkObject)
                 throw new SpawnStateException("Object is not spawned");
             }
 
-            for (int i = NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.Count - 1;
-                i > -1;
-                i--)
+            // If we made it here then we are the server and if the server is determined to already be the owner
+            // then ignore the RemoveOwnership invocation.
+            if (networkObject.OwnerClientId == NetworkManager.ServerClientId)
             {
-                if (NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects[i] == networkObject)
+                if (NetworkLog.CurrentLogLevel >= LogLevel.Normal)
                 {
-                    NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.RemoveAt(i);
+                    NetworkLog.LogWarning($"Remove ownership ignored:  {networkObject.name} is already set to the default server ownership!");
                 }
+                return;
             }
 
-            networkObject.OwnerClientIdInternal = null;
-
-            var message = new ChangeOwnershipMessage
+            // Make sure the connected client entry exists before trying to remove ownership.
+            if (NetworkManager.ConnectedClients.ContainsKey(networkObject.OwnerClientId))
```

Comment:
Seems like this could be a `TryGetValue(networkObject.OwnerClientId, out var client)` so that you don't have to index the connected clients lookup multiple times within the block